src/test/java/uk/gov/cshr/controller/signup/SignupControllerTest.java

```diff
@@ -241,8 +271,8 @@ public void shouldReturnSignupIfInviteAuthorised() throws Exception {
         mockMvc.perform(
                 get("/signup/" + code)
                         .with(CsrfRequestPostProcessor.csrf()))
-                .andExpect(status().isOk())
-                .andExpect(view().name("signup"));
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/signup/request"));
```

Comment:
This expectation and the unit test method name are not in synch. The existing method seems to be correct because of this line of the code: when(inviteService.isInviteValid(code)).thenReturn(true);
So it should remain unchanged but may need this addition: when(inviteService.isCodeExists(code)).thenReturn(true);

A new unit test method should be written with this line: when(inviteService.isCodeExpired(code)).thenReturn(false);
to prove that if the invite code is expired then user should be redirected to the signup request page.